PowerPointLabs/PowerPointLabs/ActionFramework/Common/Interface/EnabledHandler.cs

```diff
@@ -22,5 +24,26 @@ public bool Get(string ribbonId)
         }
 
         protected abstract bool GetEnabled(string ribbonId);
+        
+        protected bool HasPlaceholderInSelection()
+        {
+            var selection = GetCurrentSelection();
+            foreach (Shape shape in selection.ShapeRange)
+            {
+                if (shape.Type == Microsoft.Office.Core.MsoShapeType.msoPlaceholder)
+                {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        // Allow ActionFrameworkExtensions to get current selection
+#pragma warning disable 0618
+        protected Selection GetCurrentSelection()
```

Comment:
try to avoid this?
e.g. make all kinds of handlers inherit a class like BaseHandler,
and edit extensions ([this one](https://github.com/initialshl/PowerPointLabs/blob/dev-release/PowerPointLabs/PowerPointLabs/ActionFramework/Common/Extension/BaseActionHandlerExtensions.cs)) for this BaseHandler to access the context